engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/callactivity/CallActivityActivatingHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.handlers.callactivity;
+
+import static io.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.zeebe.engine.processor.workflow.BpmnStepContext;
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableCallActivity;
+import io.zeebe.engine.processor.workflow.handlers.CatchEventSubscriber;
+import io.zeebe.engine.processor.workflow.handlers.activity.ActivityElementActivatingHandler;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.ErrorType;
+
+public class CallActivityActivatingHandler
+    extends ActivityElementActivatingHandler<ExecutableCallActivity> {
+
+  public CallActivityActivatingHandler(CatchEventSubscriber catchEventSubscriber) {
+    super(WorkflowInstanceIntent.ELEMENT_ACTIVATED, catchEventSubscriber);
+  }
+
+  @Override
+  protected boolean handleState(BpmnStepContext<ExecutableCallActivity> context) {
+    super.handleState(context);
+
+    final var processId = context.getElement().getProcessId();
+    final var workflow = context.getStateDb().getLatestWorkflowVersionByProcessId(processId);
+
+    if (workflow == null) {
```

Comment:
I think we also need to check that the process as a single start event of type none.

Reply:
Good point. I'll add the check. Note that I need to do the check twice since a new version of the workflow could be deployed between `ACTIVATING` and `ACTIVATED`.